plugins/woocommerce/src/Internal/Admin/Notes/TrackingOptIn.php

```diff
@@ -80,9 +81,18 @@ public static function get_note() {
 	 */
 	public function opt_in_to_tracking( $note ) {
 		if ( self::NOTE_NAME === $note->get_name() ) {
+			// Get the previous value of the tracking.
+			$prev_value = get_option( 'woocommerce_allow_tracking', 'no' );
+
 			// Opt in to tracking and schedule the first data update.
 			// Same mechanism as in WC_Admin_Setup_Wizard::wc_setup_store_setup_save().
 			update_option( 'woocommerce_allow_tracking', 'yes' );
+
+			// Track woocommerce_allow_tracking_toggled in case was set as 'no' before.
+			if ( class_exists( 'WC_Tracks' ) && 'no' === $prev_value ) {
+				WC_Tracks::track_woocommerce_allow_tracking_toggled( $prev_value, 'yes', 'usage_tracking_note' );
+			}
```

Comment:
This one was a little trickier to replicate:
1. Clear any previous notes: `DELETE FROM wp_wc_admin_notes WHERE name = 'wc-admin-usage-tracking-opt-in';`
2. Disable tracking option: `wp option update woocommerce_allow_tracking no`
3. Set install time to be within week 1-4: `wp option update woocommerce_admin_install_timestamp 1733832000`
4. Run cron task `wc_admin_daily` using [WP Crontrol](https://wordpress.org/plugins/wp-crontrol/) to trigger note
5. Visit WooCommerce > Home to see note appear
6. Use this code snippet to record tracks in debug.log that are sent in API requests (WP_DEBUG_LOG should be enabled):
```php
// Log tracks.
add_filter(
	'pre_http_request',
	function ( $pre, $args, $url ) {
		if ( str_starts_with( $url, 'https://pixel.wp.com' ) ) {
			wp_parse_str( wp_parse_url( $url, PHP_URL_QUERY ), $query_vars );
			error_log( json_encode( $query_vars, JSON_PRETTY_PRINT ) );
		}
		return $pre;
	},
	10,
	3
);
```
7. Click on the activate tracking button

The tracking does get recorded. I can see the following in the logs:
```
[18-Dec-2024 15:34:53 UTC] {
    "previous_value": "no",
    "new_value": "yes",
    "context": "usage_tracking_note",
```

But somehow I get the loading state to remain stuck. I don't see any console errors:
![image](https://github.com/user-attachments/assets/dfea6433-e1b8-4c62-8413-28681cdfa9f8)

I also don't think it's related to the changes here, but maybe you know how that should be handled.

Reply:
Strange about this note. 

The only way I found was to adding actioned text. https://github.com/woocommerce/woocommerce/pull/53715/commits/05e23b26310d9df1d97ed1e669b82e7821562721